### PR TITLE
Rainbow slimes finally obtainable

### PR DESCRIPTION
**Pull request by Incoming5643**
Rainbow slimes finally added to practical play. To get one you have to raise slimes through select breeding to 100 mutation chance. Even if you're blessed with RNG This will take a minimum of 14 generations of slime. Red slime can help speed this a bit, but then you're trading plasma for time. From the default mutation chance you'd have to drop 6 of these potions.

Rainbow slimes only split into more rainbow slimes.

Rainbow slime cores spawn a random colored slime with plasma, the practical use is to allow xenobiologists to get slimes from paths they didn't breed for initially if they practiced good slime eugenics along the way. You're still at the utter mercy of RNG though.

Mostly a bragging rights reward.
#708 I've avenged you at last!

### CODE REVIEW

File: code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm [hunk 1]
```diff
@@ -618,3 +618,21 @@
...
+	result = null
+	required_reagents = list("plasma" = 1)
+	result_amount = 1
+	required_other = 1
+	required_container = /obj/item/slime_extract/rainbow
+
+/datum/chemical_reaction/slimeRNG/on_reaction(var/datum/reagents/holder)
+	feedback_add_details("slime_cores_used","[replacetext(name," ","_")]")
+	for(var/mob/O in viewers(get_turf(holder.my_atom), null))
+		O.show_message(text("<span class='danger'>Infused with plasma, the core begins to quiver and grow, and soon a new baby slime emerges from it!</span>"), 1)
```
Comment (phil235) [inline code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm]: visible_message() please.

Comment (Incoming5643) [inline code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm]: was just keeping form with how other slime cores did it, but yeah I don't see why that wouldn't work.

**Comment by PKPenguin321** [comment]
imo the amount of work to get it isn't worth the reward, but since i don't really do xenobio that much i could be dead wrong
either way it sounds pretty cool :+1:

**Comment by Supermichael777** [comment]
Beter than the amount of work being trivial for the quality of the reward. Have it do a honk-blast when injected with banana juice and I'm sold.

**Comment by Jalleo** [comment]
This could be really effective in the right(syndicate) hands

File: code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm [hunk 2]
```diff
@@ -618,3 +617,20 @@
...
+	result = null
+	required_reagents = list("plasma" = 1)
+	result_amount = 1
+	required_other = 1
+	required_container = /obj/item/slime_extract/rainbow
+
+/datum/chemical_reaction/slimeRNG/on_reaction(var/datum/reagents/holder)
+	feedback_add_details("slime_cores_used","[replacetext(name," ","_")]")
+	var/mob/living/simple_animal/slime/S = new /mob/living/simple_animal/slime
+	S.colour = pick("grey","orange", "metal", "blue", "purple", "dark purple", "dark blue", "green", "silver", "yellow", "gold", "yellow", "red", "silver", "pink", "cerulean", "sepia", "bluespace", "pyrite", "light pink", "oil", "adamantine", "black")
```
Comment (Cheridan) [inline code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm]: Could probably just do pick(typesof(mob/blahblah/slime) here for cleaner/more futureproofed code.

Comment (Incoming5643) [inline code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm]: There's actually only the one type of slime, the colour var determines everything in terms of its name/apperance/drops so unfortunately you can't do a traditional typesof.

**Comment by Kromgar** [comment]
add a blood reaction that randomly spawns a random item.

Could be anything... Guns, Magic or anything.

If it takes a long time to breed this shit add a REAL reward

**Comment by KorPhaeron** [comment]
My suggestion would be a clusterbang type thing that spits random slime cores in all directions/activates them. I'll probably make a PR for it if you don't want to do it/if that's cool with you.

**Comment by Kromgar** [comment]
Okay Kor that one is so much better. That sounds amazing

File: code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm (hunk 2 above)
Comment (Miauw62) [inline code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm]: That should be fixed sometime
Op 27-jun.-2015 03:33 schreef "Incoming5643" <notifications@github.com>:

**Pull request merged**